Source/AFError.swift

```diff
@@ -156,11 +162,15 @@ public enum AFError: Error {
         case certificatePinningFailed(host: String, trust: SecTrust, pinnedCertificates: [SecCertificate], serverCertificates: [SecCertificate])
         /// Public key pinning failed.
         case publicKeyPinningFailed(host: String, trust: SecTrust, pinnedKeys: [SecKey], serverKeys: [SecKey])
+        /// Custom evaluation failed.
+        case customEvaluationFailed(error: Error)
+        
     }
 
     /// The underlying reason the `.urlRequestValidationFailed`
     public enum URLRequestValidationFailureReason {
         case bodyDataInGETRequest(Data)
+        case customValidationFailed(error: Error)
```

Comment:
Done. However, instead of force unwrapping in `Session.swift`:
```
request.didFailToCreateURLRequest(with: error as! AFError)
```
perhaps I should instead:
```
request.didFailToCreateURLRequest(with: error.asAFError(default: .urlRequestValidationFailed(reason: .bodyDataInGETRequest(initialRequest.httpBody ?? Data()))
```

What do you think?